app/controllers/assignments_controller.rb

```diff
@@ -0,0 +1,74 @@
+class AssignmentsController < ApplicationController
+  before_filter :authorize_user!, except: [:index, :show]
+
+  def index
+    @teacher = User.find(current_user)
+    @assignments = Assignment.all.where(user_id: @teacher.id)
+  end
+
+  def show
+    @assignment = Assignment.find(params[:id])
+  end
+
+  def new
+    @teacher = User.find(current_user)
+    @assignment = Assignment.new
+  end
+
+  def create
+    @teacher = User.find(current_user)
+    @assignment = Assignment.new(assignment_params)
+    @assignment.user = @teacher
+
+    if @assignment.save
+      flash[:success] = "Assignment added successfully!"
+      redirect_to assignment_path(@assignment)
+    else
+      create_or_update_failure
+    end
+  end
+
+  def edit
+    @assignment = Assignment.find(params[:id])
+  end
+
+  def update
+    @assignment = Assignment.find(params[:id])
+    @assignment.user = current_user
+
+    if @assignment.update(assignment_params)
+      flash[:success] = "Assignment updated successfully!"
+      redirect_to assignment_path(@assignment)
+    else
+      create_or_update_failure
+    end
+  end
+
+  def destroy
+    @assignment = Assignment.find(params[:id])
+    @assignment.destroy
+    redirect_to assignments_path
+  end
+
+  private
+
+  def assignment_params
+    params.require(:assignment).permit(
+    :title,
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Indent the first parameter one step more than the start of the previous line.